src/VisualStudio/Core/Def/FindReferences/Entries/DocumentSpanEntry.cs

```diff
@@ -285,25 +285,15 @@ private static Span GetRegionSpanForReference(SourceText sourceText, TextSpan so
 
             public bool CanNavigateTo()
             {
-                if (_excerptResult.Document is SourceGeneratedDocument)
-                {
-                    var workspace = _excerptResult.Document.Project.Solution.Workspace;
-                    var documentNavigationService = workspace.Services.GetService<IDocumentNavigationService>();
-
-                    return documentNavigationService != null;
-                }
-
-                return false;
+                var workspace = _excerptResult.Document.Project.Solution.Workspace;
```

Comment:
Not sure why we wouldn't use `IDocumentNavigationService` whenever it's available from the workspace.
This is actually needed to make navigation work in interactive and semantic search workspaces where the source documents do not exist on disk.

Reply:
@jasonmalinowski PTAL